### ISSUE TITLE
[BUG] Gusy does not render templated fields not directly associated with an Operator

### ISSUE THREAD

**Issue by odaneau-astro** [pipeline-tools/gusty#68]
# Issue
Gusty does not render templated fields not directly associated with an Operator.

Some Operator fields are used in kwargs to be passed down from the Operator to the underlying hook. This is the case with the SQLExecuteQueryOperator and the hook_params field. This field is passed to the corresponding SQL Hook and allows initializing the hook with specific configurations.

Even though I have implemented a custom SQLExecuteQueryOperator to add `hook_params` to the list of templated fields, Gusty does not render the template as expected.

While Gusty will correctly render SQLExecuteQueryOperator's standard templated fields like `sql`, it does not template `hook_params`.
`hook_params` renders correctly if you use Standard Airflow Dag declaration. 

Here is the source code to reproduce:

### Folder structure
```
.
├── dags
│   ├── gusty_dag
│   │   ├── execute_query.yml
│   │   └── METADATA.yml
│   └── gusty_dag.py
├── include
│   ├── custom_operators
│   │   └── custom.py
│   └── utils.py
└── ...
```

### dags/gusty_dag/METADATA.yml
```yaml
description: "A Gusty DAG"
schedule_interval: null
default_args:
    depends_on_past: false
    start_date: !days_ago 1
```

### dags/gusty_dag/execute_query.yml
```
operator: include.custom_operators.custom.SQLExecuteQueryOperator
sql: SELECT 'gusty is causing problems';
parameters: !constant hook_params
hook_params: !constant hook_params
conn_id: snowflake_admin
```

### dags/gusty_dag.py
```
import os
from typing import Any
from gusty import create_dag
from include import utils as c

dag_dir = os.path.join(os.environ["AIRFLOW_HOME"],"dags","gusty_dag")

def constant(x: str) -> Any:
    return getattr(c, x)

macro_dict = {
    "constant": constant
}

my_dag = create_dag(dag_dir, latest_only=False, user_defined_macros=macro_dict)
```

### include/custom_operators/custom.py
```python
from typing import Sequence
from airflow.providers.common.sql.operators import sql

class SQLExecuteQueryOperator(sql.SQLExecuteQueryOperator):
    template_fields: Sequence[str] = ("sql", "parameters", "hook_params")
```

### include/utils.py
```python
hook_params = {
    "session_parameters": {
        "query_tag": (
            "{"
            "'dag_id': '{{ dag.dag_id }}', "
            "'task_id': '{{ task.task_id }}', "
            "'run_id': '{{ run_id }}', "
            "'logical_date': '{{ logical_date }}', "
            "'started': '{{ ti.start_date }}', "
            "'operator': '{{ ti.operator }}'"
            "}"
        )
    }
}
```

## Expected Behavior
Here is a standard Dag using the same custom Operator and correctly rendering `hook_params`

### dags/working_dag.py
```
from airflow.decorators import dag
from airflow.utils.dates import days_ago

from include.custom_operators.custom import (
    SQLExecuteQueryOperator,
)
from include.utils import hook_params

snowflake_conn_id = "snowflake_admin"

@dag(schedule=None, start_date=days_ago(1))
def test_hook_params_standard_airflow():
    sql_execute_query_operator = SQLExecuteQueryOperator(
        task_id="sql_execute_query_operator_task",
        conn_id=snowflake_conn_id,
        sql="SELECT '1';",
        hook_params=hook_params,
    )

test_hook_params_standard_airflow()
```

**Comment by chriscardillo** [comment]
Thanks for raising, @odaneau-astro.

gusty (lowercase "g") checks the `__init__` methods of the operator and its base operator, and does not currently consider the `template_fields` attribute.

I will cut a small release to fix this.

**Comment by chriscardillo** [comment]
Fixed in gusty v0.22.1

**Comment by odaneau-astro** [comment]
Thanks for the quick turn around. Much appreciated!
